### PR TITLE
Added general questions

### PR DESCRIPTION
**Pull request by veronvynguyen**
I questions from my recent SWE interviews, most are very basic (ex. the SSR vs CSR one)

## Type of change

- [X] New Question

# Checklist:

- [X ] My content follows the style guidelines of this project
- [X] I have performed a self-review of my own content

### CODE REVIEW

**Review by roblarsen — COMMENTED** [review]
Why do I prefer to work with individual questions? PRs like this. There are a couple of good questions here and a few I don't think make sense.

File: src/questions/general-questions.md [hunk 1]
```diff
@@ -34,3 +34,10 @@ permalink: /questions/general-questions/index.html
 * What role do you see yourself in?
 * Explain the difference between cookies, session storage, and local storage?
 * Can you explain what happens when you enter a URL into the browser?
+* Can you describe the components of a web application?
```
Comment (roblarsen) [inline src/questions/general-questions.md:37]: Not sure what this answer would actually be. I could literally write  a whole book answering this. Or someone could say "HTML + CSS + JavaScript" and then shrug.

File: src/questions/general-questions.md [hunk 2]
```diff
@@ -34,3 +34,10 @@ permalink: /questions/general-questions/index.html
...
+* What is a database transaction and how does it work?
```
Comment (roblarsen) [inline src/questions/general-questions.md:38]: Not really a front end question.

File: src/questions/general-questions.md [hunk 3]
```diff
@@ -34,3 +34,10 @@ permalink: /questions/general-questions/index.html
...
+* Describe the difference between SSR and CSR. Which one would you choose to render a web page?
```
Comment (roblarsen) [inline src/questions/general-questions.md:39]: I would like a PR with just this question. I would change the wording of the follow up to be more of a discussion point. "which one would you choose" indicates that there's **one** answer where the answer is much mroe nuanced. It's great discussion, however. Please resubmit a PR with just this question

File: src/questions/general-questions.md [hunk 4]
```diff
@@ -34,3 +34,10 @@ permalink: /questions/general-questions/index.html
...
+  * Are you familiar with static rendering?
+  * Rehydration?
+* What is your experience with git, source version control, branch development, and pull requests?
```
Comment (roblarsen) [inline src/questions/general-questions.md:42]: There are all kinds of associated questions we could add. This one is okay, but I think staying focused on F2E questions is the best way to go.

File: src/questions/general-questions.md [hunk 5]
```diff
@@ -34,3 +34,10 @@ permalink: /questions/general-questions/index.html
...
+* Describe cloud computing. What cloud platforms are you familiar with and how do you use them?
```
Comment (roblarsen) [inline src/questions/general-questions.md:43]: Again, super valid in the grand scheme of things, but... not a front end engineering question.

**Review by roblarsen — CHANGES_REQUESTED** [review]
pare down your PR to just include the CSR and SSR question and then we can hone that to be merged into the project.

File: src/questions/general-questions.md (hunk 1 above)
Comment (veronvynguyen) [reply src/questions/general-questions.md:37]: yes I agree it's very broad, but I think it's for testing your instinct and excitement overall about web development. you can start with "server and client" as components and go a little more deep into that topic, covering the basics

**Review by roblarsen — APPROVED** [review]
Thanks!

**Pull request merged**